crates/swc_ecma_minifier/src/compress/optimize/unused.rs

```diff
@@ -244,17 +244,116 @@ where
 
                     let ret = self.ignore_return_value(e);
                     if let Some(ret) = ret {
-                        *e = ret;
+                        **e = ret;
                     } else {
                         if let Some(VarDeclKind::Const | VarDeclKind::Let) = self.ctx.var_kind {
-                            *e = Null { span: DUMMY_SP }.into();
+                            **e = Null { span: DUMMY_SP }.into();
+                            return;
                         } else {
-                            *e = Expr::Invalid(Invalid { span: DUMMY_SP });
+                            **e = Expr::Invalid(Invalid { span: DUMMY_SP });
+                            return;
                         }
                     }
                 }
             }
 
+            if self.options.side_effects
+                && !v.used_as_arg
+                && !v.used_as_callee
+                && !v.indexed_with_dynamic_key
+                && !v.is_infected()
```

Comment:
Does this capture escape analysis with `return statements`, assignments to hoisted variables, etc?

Reply:
No, currently the analyzer doesn't analyze if a variable escapes